code/modules/mob/living/carbon/brain/MMI.dm

```diff
@@ -1,136 +1,218 @@
+			icon_state = "mmi-brain"
+			underlays += image(icon, null, (brainobj.icon_state in valid_brain_icon_states) ? brainobj.icon_state : "brain")
+		if(STATE_NODIODES)
+			icon_state = "mmi-nodiodes"
+			underlays += image(icon, null, (brainobj.icon_state in valid_brain_icon_states) ? brainobj.icon_state : "brain")
+		if(STATE_DIODES)
+			icon_state = "mmi-diodes"
+			underlays += image(icon, null, (brainobj.icon_state in valid_brain_icon_states) ? brainobj.icon_state : "brain")
+		if(STATE_SEALED)
+			icon_state = "mmi-sealedon"
+
+/obj/item/device/mmi/proc/update_name()
+	if(brainmob)
+		name = "[initial(name)]: [brainmob.real_name]"
+	else
+		name = initial(name)
+
+/obj/item/device/mmi/proc/set_cradle_state(var/new_state)
+	cradle_state = new_state
+	switch(cradle_state)
+		if(STATE_EMPTY)
+			extra_examine_info = "The braincase is empty, place a brain into it to begin."
+		if(STATE_BRAIN)
+			extra_examine_info = "The braincase has a brain in it, use it in-hand to flip the case over and proceed, or click on it while it's in your inactive hand to remove the brain."
+		if(STATE_NODIODES)
+			extra_examine_info = "The braincase has a brain in it and the case is flipped over, use a screwdriver in it to drive the memory suppression diodes into the brain, or use it in-hand to flip the case back and prepare the brain for removal."
+		if(STATE_DIODES)
+			extra_examine_info = "The braincase's memory suppression diodes have been inserted, use a welding tool on it to seal it permanently and prepare it for use, or use a screwdriver on it to undo the diodes."
+		if(STATE_SEALED)
+			extra_examine_info = "The braincase is sealed and ready for use. The only thing that will undo the seal is a circular saw, and that will destroy the brain inside."
+
+/obj/item/device/mmi/examine(mob/user, distance)
+	. = ..()
+	if(extra_examine_info)
+		to_chat(user, SPAN_NOTICE(extra_examine_info))
+
+/obj/item/device/mmi/attackby(obj/item/I, mob/user)
+	switch(cradle_state)
+		if(STATE_EMPTY)
+			if(!brainmob && istype(I, /obj/item/organ/internal/brain)) //Time to stick a brain in it --NEO
+				var/obj/item/organ/internal/brain/B = I
+				if(!B.can_prepare)
+					to_chat(user, SPAN_WARNING("\The [B] is incompatible with [src]!"))
+					return
+				if(B.health <= 0)
+					to_chat(user, SPAN_WARNING("That brain is well and truly dead."))
+					return
+				else if(!B.brainmob)
+					to_chat(user, SPAN_WARNING("You aren't sure where this brain came from, but you're pretty sure it's a useless brain."))
+					return
+
+				user.visible_message("<b>[user]</b> places \the [B] into \the [src].", SPAN_NOTICE("You place \the [B] into \the [src]."))
+
+				brainmob = B.brainmob
+				B.brainmob = null
+				brainmob.forceMove(src)
+				brainmob.container = src
+				brainmob.stat = CONSCIOUS
+				dead_mob_list -= brainmob //Update dem lists
+				living_mob_list += brainmob
+
+				brainobj = B
+				user.drop_from_inventory(brainobj, src)
+
+				set_cradle_state(STATE_BRAIN)
```

Comment:
Every time you call this proc you call update_icon() right after it, and usually update_name(). Couldn't you roll both of those into set_cradle_state? Otherwise looks good

Reply:
yeah, update_name only needs to happen when a brain gets added or removed, but update_icon can be rolled into set_cradle_state, good spot